config/v1/types_network.go

```diff
@@ -67,6 +73,36 @@ type ClusterNetworkEntry struct {
 	HostPrefix uint32 `json:"hostPrefix"`
 }
 
+// ExternalIPConfig specifies some IP blocks relevant for the ExternalIP field
+// of a Service resource.
+type ExternalIPConfig struct {
+	// policy is a set of restrictions applied to the ExternalIP field.
+	// If nil, any value is allowed for an ExternalIP. If the empty/zero
+	// policy is supplied, then ExternalIP is not allowed to be set.
+	// +optional
+	Policy *ExternalIPPolicy `json:"policy,omitempty"`
+
+	// autoAssignCIDRs is a list of CIDRs from which to automatically assign
+	// Service.ExternalIP. These are assigned when the service is of type
+	// LoadBalancer. In general, this is only useful for bare-metal clusters.
+	// In Openshift 3.x, this was misleadingly called "IngressIPs".
+	// Currently, only one entry may be provided.
```

Comment:
One entry and you pass array?

Reply:
We've learned that these sorts of things often get expanded. Right now, the controller only supports a single CIDR pool, but there's no technical reason that can't change. No sense in blocking ourselves in with an unnecessarily restrictive API.